src/Simplex/Messaging/Agent.hs

```diff
@@ -1944,16 +2056,28 @@ processSMPTransmission c@AgentClient {smpClients, subQ} (tSess@(_, srv, _), v, s
         -- mark queue as Secured and to start sending messages to it
         qUseMsg :: NonEmpty ((SMPServer, SMP.SenderId), Bool) -> Connection 'CDuplex -> m ()
         -- NOTE: does not yet support the change of the primary status during the rotation
-        qUseMsg ((addr, _primary) :| _) (DuplexConnection _ _ sqs) =
-          case findQ addr sqs of
-            Just sq' -> do
-              logServer "<--" c srv rId $ "MSG <QUSE> " <> logSecret (snd addr)
-              withStore' c $ \db -> setSndQueueStatus db sq' Secured
-              let sq'' = (sq' :: SndQueue) {status = Secured}
-              -- sending QTEST to the new queue only, the old one will be removed if sent successfully
-              void $ enqueueMessages c cData [sq''] SMP.noMsgFlags $ QTEST [addr]
-              notify . SWITCH QDSnd SPConfirmed $ connectionStats conn
-            _ -> qError "QUSE: queue address not found in connection"
+        qUseMsg ((addr, _primary) :| _) (DuplexConnection _ rqs sqs) =
+          case switchingSQ sqs of
+            Just replaced -> do
+              checkSQSwchStatus replaced SSSendingQKEY
+              replaced' <- withStore' c $ \db -> setSndQueueSwitchStatus db replaced (Just SSReceivedQUSE)
```

Comment:
why do we need the status in the beginning of the operation?

it's never checked for, not sent to the client, and if the connection stays in it it certainly can't progress switching. 

There are other cases like that too.